dali/kernels/imgproc/sampler.h

```diff
@@ -30,165 +30,183 @@ namespace kernels {
 
 struct BorderClamp {};
 
-template <DALIInterpType interp, typename In>
+template <DALIInterpType interp, int spatial_ndim, typename In>
```

Comment:
Why int and not unsigned?

Reply:
It's always int everywhere. TensorShape, vec, mat and all the kernels.

Reply:
@JanuszL We agreed some time ago to use `int` on template arguments, to avoid problems with deducing the arguments

Reply:
Sure. Fine for me.